SU-SWS/decanter
# 215 Logo lockup

# READY FOR REVIEW

# Summary
- Lockup component with Stanford wordmark logo and the site (unit/department) name. 

# Needed By (Date)
- N/A

# Urgency
- N/A

# Steps to Test

1. Pull this branch
2. Run grunt styleguide
3. Check if the lockup with the individual line variants match the design in the Figma file [Stanford Design Library "Header, Footer & Navigation" page](https://www.figma.com/file/Kmd4utmJFPRMVeCFEEBQhLtx/Stanford-Design-Library?node-id=1%3A23), some of the variants in the [Stanford Open Framework site lockup guide](https://docs.google.com/document/d/14_92mZHRocYWwFQfMWdTz5kQpT1QoSGj4AazAsRM5mc/edit?usp=sharing), and the [Identity toolkit page on department lockups](https://identity.stanford.edu/department-branding.html#using-dept-sig).
4. Check behavior for smaller breakpoints. 
5. Try changing values in lockup.json (eg, with or without line 2/5, different variants for line 1/5) and see if interplay works as expected (eg, horizontal line on mobile disappear when there is no line 5, variant of line 1 affects font size of line 2 etc).

# Affected Projects or Products
- Decanter

# Associated Issues and/or People
- https://github.com/SU-SWS/decanter/issues/215
- Only including line 1, 2, and 5 in the site name for this version. Will add support for adding line 3 and 4 in future release.
- Added conditionals in twig file to check for which variant of the lockup to use (default one-line or the two-line variant) - please let me know if this is a good approach.
- I started with font-size: 30px (same as in Figma file [Stanford Design Library "Header, Footer & Navigation" page](https://www.figma.com/file/Kmd4utmJFPRMVeCFEEBQhLtx/Stanford-Design-Library?node-id=1%3A23) - the word "Biology" next to wordmark) for line 1 regular variant, and eyeballed the rest of the font sizes based on the [Stanford Open Framework site lockup guide](https://docs.google.com/document/d/14_92mZHRocYWwFQfMWdTz5kQpT1QoSGj4AazAsRM5mc/edit?usp=sharing).
- With/without line 5 (at the bottom) isn't a variant since the grid simply collapses when it's not there.
- The ordering of the elements in twig file is: wordmark, pipe, line 5, line 1, line 2 since I need to use the adjacent CSS selector + to have the line show up only when line 5 is present on mobile.
- @sherakama (just FYI in case you want to comment on anything since the way I coded this was a bit experimental)
